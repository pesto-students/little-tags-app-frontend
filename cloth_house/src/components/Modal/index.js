import React from 'react';
import { Modal, Button } from 'antd';
import propTypes from "prop-types";
export default class CustomModal extends React.Component {
  state = {
    loading: false,
    visible: false,
  };

  showModal = () => {
    this.setState({
      visible: true,
    });
  };

  handleOk = () => {
    this.setState({ loading: true });
    setTimeout(() => {
      this.setState({ loading: false, visible: false });
    }, 3000);
  };

  handleCancel = () => {
    this.setState({ visible: false });
  };

  render() {

  console.log("Props======>",this.props);  
    const { visible, loading } = this.state;
    const defaultFooter=[
        <Button key="back" onClick={this.props.handleCancel||this.handleCancel}>
          Return
        </Button>,
        <Button key="submit" type="primary" loading={loading} onClick={this.handleOk}>
          Submit
        </Button>,
      ]
 let body="";
 if(this.props.callFrom==="order"){
body=this.props.body
 }else{
   body=   this.props.intl&&
   this.props.intl.formatMessage({id:this.props.body&&this.props.body.toString()})
  }

 
 
      return (
      <>
        <Modal
          visible={this.props.visible||visible}
          title={this.props.title}
          onOk={this.props.handleOk||this.handleOk}
          footer={this.props.footer||defaultFooter}
          onCancel={this.props.handleCancel||this.handleCancel}
        >{body}
        </Modal>
      </>
    );
  }
}

CustomModal.propTypes={
 loading:propTypes.bool,
 visible:propTypes.bool,
 handleOk:propTypes.func,
 footer:propTypes.array,
 title:propTypes.string   
}