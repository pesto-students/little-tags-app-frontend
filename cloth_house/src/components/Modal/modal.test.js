import createComponentWithIntl from '../../utils/helper_snapshot'
import Modal from "./index"

it('Agreement Modal Snapshot', () => {
    const component = createComponentWithIntl(<Modal />);
    let tree = component.toJSON()
  
    expect(tree).toMatchSnapshot()
  
  });