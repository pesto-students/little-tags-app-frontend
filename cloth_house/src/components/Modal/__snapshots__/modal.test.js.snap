// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Agreement Modal Snapshot 1`] = `null`;

exports[`App Snapshot 1`] = `null`;
